app/components/Views/AccountActions/AccountActions.tsx

```diff
@@ -78,14 +75,10 @@ const AccountActions = () => {
     return { KeyringController, PreferencesController };
   }, []);
 
-  const existingKeyrings = useSelector(selectKeyrings);
-
-  const keyringId = useMemo(() => {
-    const keyring = existingKeyrings.find((kr) =>
-      kr.accounts.includes(selectedAccount.address.toLowerCase()),
-    );
-    return keyring?.metadata.id;
-  }, [existingKeyrings, selectedAccount.address]);
+  const keyringId = useMemo(
+    () => selectedAccount.options.entropySource,
+    [selectedAccount.options.entropySource],
+  );
```

Comment:
This wouldn't work for hd accounts because they don't have entropySource. 

Maybe we should check for both.
```suggestion
  const existingKeyrings = useSelector(selectKeyrings);

  const keyringId = useMemo(() => {
    const keyring = existingKeyrings.find((kr) =>
      kr.accounts.includes(selectedAccount.address.toLowerCase()),
    );
    return isFirstPartySnapAccount ? selectedAccount.options.entropySource : keyring?.metadata.id; 
  }, [existingKeyrings, selectedAccount, isFirstPartySnapAccount]);
  
  
  
```

Reply:
Nevermind, this is being manually added in the accounts controller.

Reply:
It seems like the entropy source is automatically added by the acounts controller on the latest version. Confirmed that the entropy source for an HD account is the keyring.metadata.id.

[Code](https://github.com/MetaMask/core/blob/main/packages/accounts-controller/src/AccountsController.ts#L839-L843)

```
     patch.added.push({
            address,
            type: keyring.type,
            // Automatically injects `entropySource` for HD accounts only.
            options:
              keyring.type === KeyringTypes.hd
                ? { entropySource: keyring.metadata.id }
                : {},
          });
```